lib/new_relic/agent/instrumentation/roda.rb

```diff
@@ -0,0 +1,40 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'roda/instrumentation'
+require_relative 'roda/chain'
+require_relative 'roda/prepend'
+
+DependencyDetection.defer do
+  named :roda
+
+  depends_on do
+    defined?(Roda) &&
+      Gem::Version.new(Roda::RodaVersion) >= Gem::Version.new('3.19.0') &&
+      Roda::RodaPlugins::Base::ClassMethods.private_method_defined?(:build_rack_app) &&
+      Roda::RodaPlugins::Base::InstanceMethods.method_defined?(:_roda_handle_main_route)
+  end
+
+  executes do
+    # These requires are inside an executes block because they require rack, and
+    # we can't be sure that rack is available when this file is first required.
```

Comment:
Hmm I see what you're saying. [Moved these ](https://github.com/newrelic/newrelic-ruby-agent/pull/2144/files#diff-8067a27ede8ceb13b66d3b4c1389e9cb9994ec127bc1820ce451975d6c156aecR5-R6)to where roda instrumentation logic lives.

Reply:
Backup - our tests didn't like them existing in roda instrumentation logic. I've [moved](https://github.com/newrelic/newrelic-ruby-agent/pull/2144/files#diff-7f80c12f838c8bc306fce6b66036c7760f13728f57ce6622f1de94a966c40c6aR18-R19) them back so that I can just call these requires once vs putting them in the chain and prepend files.